src/main/java/org/terasology/multiBlock2/system/MultiBlockServerSystem.java

```diff
@@ -111,6 +115,28 @@ public void update(float delta) {
             }
         }
     }
+    @ReceiveEvent
+    public void onStructureSpawn(SendRegionEvent event, EntityRef entity) {
+        ArrayList<Vector3i> blocks = new ArrayList<>();
+        Vector3f location = new Vector3f();
+        for (SpawnBlockRegionsComponent.RegionToFill region : event.regions) {
+            for (Vector3ic pos : region.region) {
+                blocks.add(new Vector3i(pos));
+            }
+        }
+        entity.getComponent(LocationComponent.class).getWorldPosition(location);
+        MultiBlockDefinition definition = new DefaultMultiBlockDefinition("Structure",
+                new Vector3i((int) location.x(), (int) location.y(), (int) location.z()), blocks);
```

Comment:
I would have expected to see this within the ST module, without the need for MB to know about ST at all.